modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -363,8 +371,15 @@
 (defmethod ->rvalue :concat    [[_ & args]] {"$concat" (mapv ->rvalue args)})
 (defmethod ->rvalue :substring [[_ & args]] {"$substrCP" (mapv ->rvalue args)})
 
-(defmethod ->rvalue :+ [[_ & args]] {"$add" (mapv ->rvalue args)})
-(defmethod ->rvalue :- [[_ & args]] {"$subtract" (mapv ->rvalue args)})
+(defmethod ->rvalue :+ [[_ & [lhs rhs :as args]]]
+  (cond
+    (interval? lhs) (date-op "$dateAdd" rhs lhs)
+    (interval? rhs) (date-op "$dateAdd" lhs rhs)
+    :else {"$add" (mapv ->rvalue args)}))
+(defmethod ->rvalue :- [[_ & [lhs rhs :as args]]]
+  (if (interval? rhs)
+    (date-op "$dateSubtract" lhs rhs)
+    {"$subtract" (mapv ->rvalue args)}))
```

Comment:
```suggestion

(defmethod ->rvalue :+ [[_ & [lhs rhs :as args]]]
  (cond
    (interval? lhs) (date-op "$dateAdd" rhs lhs)
    (interval? rhs) (date-op "$dateAdd" lhs rhs)
    :else {"$add" (mapv ->rvalue args)}))
    
(defmethod ->rvalue :- [[_ & [lhs rhs :as args]]]
  (if (interval? rhs)
    (date-op "$dateSubtract" lhs rhs)
    {"$subtract" (mapv ->rvalue args)}))
    
```

Please add some newlines here, having no lines between them stops making sense when the forms are more than one line